BungeeCord-Patches/0064-Adventure.patch

```diff
@@ -0,0 +1,1668 @@
+From 93bb936ae38b16084d021f522e461a1fcba439f0 Mon Sep 17 00:00:00 2001
+From: Callum Seabrook <callum.seabrook@prevarinite.com>
+Date: Wed, 28 Apr 2021 17:02:18 +0100
+Subject: [PATCH] Adventure
+
+
+diff --git a/api/pom.xml b/api/pom.xml
+index ca1ed881..c53ad692 100644
+--- a/api/pom.xml
++++ b/api/pom.xml
+@@ -18,6 +18,28 @@
+     <name>Waterfall-API</name>
+     <description>API implemented by the Elastic Portal Suite</description>
+ 
++    <!-- Waterfall start - add snapshot repository for Adventure's BungeeCord serializer and add BOM -->
++    <repositories>
++        <!-- Waterfall - Add repository required for Adventure's BungeeCord serializer -->
++        <repository>
++            <id>sonatype-oss-snapshots</id>
++            <url>https://oss.sonatype.org/content/repositories/snapshots/</url>
++        </repository>
++    </repositories>
++
++    <dependencyManagement>
++        <dependencies>
++            <dependency>
++                <groupId>net.kyori</groupId>
++                <artifactId>adventure-bom</artifactId>
++                <version>4.7.0</version>
++                <type>pom</type>
++                <scope>import</scope>
++            </dependency>
++        </dependencies>
++    </dependencyManagement>
++    <!-- Waterfall end -->
++
+     <dependencies>
+         <dependency>
+             <groupId>io.github.waterfallmc</groupId>
+@@ -82,5 +104,24 @@
+             <artifactId>slf4j-api</artifactId>
+             <version>1.7.25</version>
+         </dependency>
++        <!-- Waterfall start - Add Adventure -->
++        <dependency>
++            <groupId>net.kyori</groupId>
++            <artifactId>adventure-api</artifactId>
++        </dependency>
++        <dependency>
++            <groupId>net.kyori</groupId>
++            <artifactId>adventure-text-serializer-gson</artifactId>
++        </dependency>
++        <dependency>
++            <groupId>net.kyori</groupId>
++            <artifactId>adventure-text-serializer-legacy</artifactId>
++        </dependency>
++        <dependency>
++            <groupId>net.kyori</groupId>
++            <artifactId>adventure-text-serializer-bungeecord</artifactId>
++            <version>4.0.0-SNAPSHOT</version>
++        </dependency>
```

Comment:
We really do not need to pull this in. We can do this natively instead.

Reply:
How exactly do you want me to do this? Write this in myself? Or do you have other ideas?

Reply:
Guess I could copy this into the API itself (in the `io.github.waterfallmc.waterfall.adventure` package), but I'm not sure this is the greatest idea, as it means that we have to manually update this every time you update it on your end, if you see my thinking.

Reply:
We can do this natively instead.